tests/test_task_api.py

```diff
@@ -371,3 +371,65 @@ async def test_task_collection_api_failure(
         assert data["status"] == "fail"
         assert data["log"]  # This is because of verbose=True
         assert "fail" in data["log"]
+
+
+async def test_non_python_task(
+    client,
+    MockCurrentUser,
+    testdata_path,
+    project_factory,
+    dataset_factory,
+    resource_factory,
+    tmp777_path,
+):
+    async with MockCurrentUser(persist=True) as user:
+        project_dir = tmp777_path / "test"
+        project = await project_factory(user, project_dir=str(project_dir))
+        payload = {"name": "WF", "project_id": project.id}
+        res = await client.post("api/v1/workflow/", json=payload)
+        workflow = res.json()
+        debug(workflow)
+        assert res.status_code == 201
+
+        task_dict = dict(
+            name="non-python",
+            source="custom task",
+            command=(
+                f"sh {str(testdata_path)}/issue189.sh "
+                "--json hello --metadata-out world"
+            ),
+            input_type="zarr",
+            output_type="zarr",
+        )
+        task_create = TaskCreate(**task_dict)
+        res = await client.post("api/v1/task/", json=task_create.dict())
+        assert res.status_code == 201
+        task = res.json()
+        res = await client.post(
+            f"api/v1/workflow/{workflow['id']}/add-task/",
+            json=dict(task_id=task["id"]),
+        )
+        assert res.status_code == 201
+
+        input_dataset = await dataset_factory(
+            project, name="input", type="zarr", read_only=False
+        )
+        input_dataset_id = input_dataset.id
+        output_dataset = await dataset_factory(
+            project, name="output", type="zarr", read_only=False
+        )
+        output_dataset_id = output_dataset.id
+
+        await resource_factory(path="/tmp", dataset=output_dataset)
+
+        payload = dict(
+            project_id=project.id,
+            input_dataset_id=input_dataset_id,
+            output_dataset_id=output_dataset_id,
+            workflow_id=workflow["id"],
+            overwrite_input=False,
+        )
+        debug(payload)
+        res = await client.post("/api/v1/project/apply/", json=payload)
+        debug(res.json())
+        assert res.status_code == 202
```

Comment:
We should add an assertion about the output. For instance we could modify the task so that it reads the input file (`--json`) and writes it to the output file (`--metadata-out`), and then check that the output file is equal to the input file.